From c0c98376757eaca85097f0fa3365641fcdc62af1 Mon Sep 17 00:00:00 2001
From: AKASHI Takahiro <takahiro.akashi@linaro.org>
Date: Fri, 8 May 2020 14:51:21 +0900
Subject: [PATCH 1/1] efi_loader: variable: check a return value of
 uuid__str_to_bin()

The only error case is that a given UUID is in wrong format.
So just return EFI_INVALID_PARAMETER here.

Reported-by: Coverity (CID 300333)
Signed-off-by: AKASHI Takahiro <takahiro.akashi@linaro.org>
Reviewed-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_loader/efi_variable.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/lib/efi_loader/efi_variable.c b/lib/efi_loader/efi_variable.c
index 60c1201757..10892684d1 100644
--- a/lib/efi_loader/efi_variable.c
+++ b/lib/efi_loader/efi_variable.c
@@ -767,7 +767,10 @@ static efi_status_t parse_uboot_variable(char *variable,
 	/* guid */
 	c = *(name - 1);
 	*(name - 1) = '\0'; /* guid need be null-terminated here */
-	uuid_str_to_bin(guid, (unsigned char *)vendor, UUID_STR_FORMAT_GUID);
+	if (uuid_str_to_bin(guid, (unsigned char *)vendor,
+			    UUID_STR_FORMAT_GUID))
+		/* The only error would be EINVAL. */
+		return EFI_INVALID_PARAMETER;
 	*(name - 1) = c;
 
 	/* attributes */
-- 
2.26.2

